storybookjs/storybook
# Nextjs: Improve support for Windows-style paths

Added more portable path handling code to better support win32-style paths. Prior to this PR, attempting to use fonts via `next/font/local' in Storybook on a Windows machine would result in broken font resource paths.

Closes #21968

## What I did

Added more portable path handling code to better support win32-style paths. As demonstrated by #21968, a bug existed with the current way that paths were being translated between `next/font` and Storybook. This PR utilizes more portable path handling logic and sanitizes win32-style paths to the Unix-style paths expected in standard web URIs.

## How to test

1. Open a story that uses fonts loaded via `next/font` on a Windows-based machine.
2. Check the `@font-face` declarations that are generated into the story and ensure that the paths are correct, and that the fonts are being loaded and are available in the story.

## Checklist

      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers


`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`
